ydb/core/tx/schemeshard/schemeshard_info_types.cpp

```diff
@@ -186,6 +191,23 @@ void TSubDomainInfo::CountDiskSpaceQuotas(IQuotaCounters* counters, const TDiskS
     if (softDelta != 0) {
         counters->ChangeDiskSpaceSoftQuotaBytes(softDelta);
     }
+    for (const auto& [poolKind, newPoolQuotas] : next.StoragePoolsQuotas) {
+        const auto* oldPoolQuotas = prev.StoragePoolsQuotas.FindPtr(poolKind);
+        i64 delta = i64(newPoolQuotas.SoftQuota) - (oldPoolQuotas ? i64(oldPoolQuotas->SoftQuota) : 0);
+        if (delta != 0) {
```

Comment:
Expression `i64 delta = i64(a) - i64(b)` will have unexpected result if `a > INT64_MAX` and `b < INT64_MAX` (and vice versa). So all that delta calculations effectively means that quota values must not be set to values greater than INT64_MAX. And this is something that users would never expect.

I think its time to right the wrong and change all delta expressions to `i64 delta = a - b`. It is the least that we could do here.